cmd/cluster/kubevirt/create_test.go

```diff
@@ -0,0 +1,112 @@
+package kubevirt
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	apifixtures "github.com/openshift/hypershift/api/fixtures"
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/cmd/cluster/core"
+)
+
+func Test_ApplyPlatformSpecificValues(t *testing.T) {
+	tests := map[string]struct {
+		argsOpts             core.CreateOptions
+		expectedPlatformOpts apifixtures.ExampleOptions
+		expectedError        bool
+	}{
+		"should success configuring additional networks including default one": {
+			argsOpts: core.CreateOptions{
+				InfraID: "infra1",
+				KubevirtPlatform: core.KubevirtPlatformCreateOptions{
+					ServicePublishingStrategy: "Ingress",
+					Cores:                     1,
+					RootVolumeSize:            16,
+					AdditionalNetworks: []string{
+						"name:ns1/nad1",
+						"name:ns2/nad2",
+					},
+					AttachDefaultNetwork: true,
+				},
+			},
+			expectedPlatformOpts: apifixtures.ExampleOptions{
+				InfraID: "infra1",
+				Kubevirt: &apifixtures.ExampleKubevirtOptions{
+					ServicePublishingStrategy: "Ingress",
+					Cores:                     1,
+					RootVolumeSize:            16,
+					BaseDomainPassthrough:     true,
+					AdditionalNetworks: []hyperv1.KubevirtNetwork{
+						{
+							Name: "ns1/nad1",
+						},
+						{
+							Name: "ns2/nad2",
+						},
+					},
+					AttachDefaultNetwork: true,
+				},
+			},
+		},
+		"should success configuring additional networks excluding default network": {
+			argsOpts: core.CreateOptions{
+				InfraID: "infra1",
+				KubevirtPlatform: core.KubevirtPlatformCreateOptions{
+					ServicePublishingStrategy: "Ingress",
+					Cores:                     1,
+					RootVolumeSize:            16,
+					AdditionalNetworks: []string{
+						"name:ns1/nad1",
+						"name:ns2/nad2",
+					},
+					AttachDefaultNetwork: false,
+				},
+			},
+			expectedPlatformOpts: apifixtures.ExampleOptions{
+				InfraID: "infra1",
+				Kubevirt: &apifixtures.ExampleKubevirtOptions{
+					ServicePublishingStrategy: "Ingress",
+					Cores:                     1,
+					RootVolumeSize:            16,
+					BaseDomainPassthrough:     true,
+					AdditionalNetworks: []hyperv1.KubevirtNetwork{
+						{
+							Name: "ns1/nad1",
+						},
+						{
+							Name: "ns2/nad2",
+						},
+					},
+					AttachDefaultNetwork: false,
+				},
+			},
+		},
+		"should fail excluding default network without additional ones": {
+			argsOpts: core.CreateOptions{
+				InfraID: "infra1",
+				KubevirtPlatform: core.KubevirtPlatformCreateOptions{
+					ServicePublishingStrategy: "Ingress",
+					Cores:                     1,
+					RootVolumeSize:            16,
```

Comment:
are these values really required ? I would omit them - afaiu, we only care about the `AttachDefaultNetwork` flag here.

Reply:
It fails if we do not pass them.